src/test/java/com/users/service/UserServiceTest.java

```diff
@@ -0,0 +1,244 @@
+        user.setDateOfBirth(userDto.getDateOfBirth());
+        user.setPrimaryEmail(userDto.getPrimaryEmail());
+        user.setPhoneNumber(userDto.getPhoneNumber());
+
+        when(userMapper.toEntity(userDto)).thenReturn(user);
+        when(userMapper.toDto(user)).thenReturn(userDto);
+        when(userRepository.save(any(User.class))).thenReturn(user);
+
+        UserDto createdDto = userService.create(userDto);
+
+        assertThat(createdDto.getDateOfBirth()).isSameAs(userDto.getDateOfBirth());
+        assertThat(createdDto.getPrimaryEmail()).isSameAs(userDto.getPrimaryEmail());
+        assertThat(createdDto.getPhoneNumber()).isSameAs(userDto.getPhoneNumber());
+
+        verify(userRepository).save(user);
+    }
+
+    @Test(expected = EntityAlreadyExistsException.class)
+    public void whenCreateUser_withPrimaryEmailInSecondaryEmails_shouldThrowException() {
+        List<UserDto> usersDto = UserUtilityTest.createUsersDto(1);
+        UserDto userDto = usersDto.get(0);
+        userDto.setSecondaryEmails(Arrays.asList(userDto.getPrimaryEmail(), "another@yahoo.com"));
+
+        userService.create(userDto);
+    }
+
+    @Test(expected = EntityAlreadyExistsException.class)
+    public void whenCreateUser_withSecondaryEmailAlreadyExistingAsPrimary_shouldThrowException() {
+        List<UserDto> usersDto = UserUtilityTest.createUsersDto(1);
+        UserDto userDto = usersDto.get(0);
+        userDto.setSecondaryEmails(List.of("existing@yahoo.com"));
+
+        when(userRepository.existsByPrimaryEmail("existing@yahoo.com")).thenReturn(true);
+
+        userService.create(userDto);
+    }
+
+    @Test
+    public void whenGivenId_shouldDeleteUser_ifFound() {
+        List<User> usersEntity = UserUtilityTest.createUsersEntity(1);
+        User userEntity = usersEntity.get(0);
+
+        when(userRepository.findById(userEntity.getId())).thenReturn(Optional.of(userEntity));
+
+        userService.deleteUser(userEntity.getId());
+
+        verify(userRepository).deleteById(userEntity.getId());
+    }
+
+    @Test(expected = EntityNotFoundException.class)
+    public void whenDeleteUser_shouldThrowException_ifUserNotFound() {
+        when(userRepository.findById(anyLong())).thenReturn(Optional.empty());
+
+        userService.deleteUser(99L);
+    }
+
+    @Test
+    public void shouldReturnAllUsers() {
+        UserCriteriaEasy criteria = new UserCriteriaEasy();
+        Pageable pageable = PageRequest.of(0, 10);
+
+        List<UserDto> usersDto = UserUtilityTest.createUsersDto(2);
+        UserDto userDto1 = usersDto.get(0);
+        UserDto userDto2 = usersDto.get(1);
+
+        User user1 = new User();
+        user1.setDateOfBirth(userDto1.getDateOfBirth());
+        user1.setPrimaryEmail(userDto1.getPrimaryEmail());
+        user1.setPhoneNumber(userDto1.getPhoneNumber());
+        user1.setSecondaryEmails(userDto1.getSecondaryEmails());
+
+        User user2 = new User();
+        user2.setDateOfBirth(userDto2.getDateOfBirth());
```

Comment:
mai e o varianta care nu este considerata gresita, la clasa de mapper sa o aduci in context ul acestei clase de test, gen sa nu o consideri un serviciu extern, ca sa nu mai populezi aceste teste cu mapare de la entity la dto sau invers